pkgs/os-specific/linux/dcgm/default.nix

```diff
@@ -85,13 +77,13 @@ let
 # C.f. https://github.com/NVIDIA/DCGM/blob/7e1012302679e4bb7496483b32dcffb56e528c92/dcgmbuild/build.sh#L22
 in gcc11Stdenv.mkDerivation rec {
```

Comment:
Ok, I suspect this means that you have to override all other (dynamically linked) dependencies to use the same stdenv, just like you did for `yaml-cpp`. Otherwise they might link whatever libstdc++ is default in nixpkgs (gcc13). If DCGM links a libstdc++ too (I guess it does, otherwise why override yaml-cpp), it'll load the one coming with gcc11 which won't have the symbols needed by the dependencies (e.g. by yaml-cpp)...

I think a better solution is to make DGCM link the newer libstdc++. This is what we do for cudaPackages: https://github.com/NixOS/nixpkgs/blob/bfaf0a1feb46a3ef8b17a195d52ee8d4c029e28a/pkgs/development/cuda-modules/backend-stdenv.nix#L14 (disclaimer: `useLibsFrom` might get removed in the future; it's ok, we'll figure out a migration plan)

This way you won't have to override `yaml-cpp` or any other packages stdenv-wise

Reply:
This translates into `useLibsFrom stdenv gcc11Stdenv`